### ISSUE TITLE
Replace all `torch.Tensor` by `torch.tensor`

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#2527]
## 📚 Documentation

@sdesrozis  it is incorrect to create tensors with `torch.Tensor` and should be replaced by `torch.tensor`. 

https://github.com/pytorch/ignite/blob/3b821e27ec41fffbbe5e43658798040f7024ba0b/ignite/metrics/accuracy.py#L125-L134

This way we wont need neither cast to `.long` : `y_true = torch.Tensor([2, 0, 2, 1, 0, 1]).long()`

https://github.com/pytorch/ignite/blob/3b821e27ec41fffbbe5e43658798040f7024ba0b/ignite/metrics/accuracy.py#L146
